### PR TITLE
fix compile errors with rmw_connext_dynamic_cpp

### PR DESCRIPTION
**Pull request by dirk-thomas**
Replaces #210.

### CODE REVIEW

File: rmw_connext_dynamic_cpp/src/functions.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@
 #include "rmw/types.h"
 // This header is in the rosidl_typesupport_connext_cpp package and
 // is in the include/rosidl_typesupport_connext_cpp/impl folder.
```
Comment (Karsten1987) [inline rmw_connext_dynamic_cpp/src/functions.cpp]: is that comment still valid?

**Comment by dirk-thomas** [comment]
It currently still fails later on downstream packages - will get that sorted out soon...

File: rmw_connext_dynamic_cpp/src/functions.cpp (hunk 1 above)
Comment (dirk-thomas) [inline rmw_connext_dynamic_cpp/src/functions.cpp]: No, will remove the comment.

**Comment by dirk-thomas** [comment]
Ready for review.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/2082/)
* Mac OS [Build Status](http://ci.ros2.org/job/ci_osx/1608/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/2100/)

**Pull request merged**